packages/flame/lib/src/experimental/layout_component.dart

```diff
@@ -0,0 +1,146 @@
+import 'package:flame/components.dart';
+import 'package:flame/experimental.dart';
+import 'package:flame/src/game/notifying_vector2.dart';
+import 'package:flutter/widgets.dart';
+
+enum Direction { horizontal, vertical }
+
+/// Super class for [RowComponent] and [ColumnComponent]
+/// A relayout is performed when
+///  - a change in this component's children takes place
+///  - the size of this component or a parent changes
+///  - the [gap] parameter is changed
+abstract class LayoutComponent extends PositionComponent {
```

Comment:
Is this class intended to be user-facing? That is, do we anticipate a need for a derived class that is neither RowComponent nor ColumnComponent? If not, then this could be declared `@internal` and un-exported.

Reply:
It's all experimental at the moment, so we just don't know yet.